src/renderer/commands/drawObject.ts

```diff
@@ -0,0 +1,118 @@
+// tslint:disable-next-line:import-name
+import REGL = require('regl');
+
+// tslint:disable:no-unsafe-any
+
+interface Uniforms {
+    projection: REGL.Mat4;
+    view: REGL.Mat4;
+    model: REGL.Mat4;
+    numLights: number;
+    lightPositions: REGL.Vec3[];
+    lightColors: REGL.Vec3[];
+    lightIntensities: number[];
+}
+
+interface Attributes {
+    position: REGL.Vec3;
+    normal: REGL.Vec3;
+    color: REGL.Vec3;
+}
+
+/*
+ * All the information needed to be able to draw an object to the screen
+ */
+export interface DrawObjectProps {
+    model: REGL.Mat4;
+    cameraTransform: REGL.Mat4;
+    projectionMatrix: REGL.Mat4;
+    positions: REGL.Vec3[];
+    normals: REGL.Vec3[];
+    colors: REGL.Vec3[];
+    indices: number[];
+}
+
+/*
+ * Shader to draw a single object with Phong shading to the screen
+ */
+export function drawObject(
+    regl: REGL.regl
+): REGL.DrawCommand<REGL.DefaultContext, DrawObjectProps> {
+    return regl<Uniforms, Attributes, DrawObjectProps>({
+        vert: `
+            precision mediump float;
+
+            attribute vec3 position;
+            attribute vec3 normal;
+            attribute vec3 color;
+
+            uniform mat4 projection;
+            uniform mat4 view;
+            uniform mat4 model;
+
+            varying vec3 vertexPosition;
+            varying vec3 vertexNormal;
+            varying vec3 vertexColor;
+
+            void main() {
+                vertexPosition = (view * model * vec4(position, 1.0)).xyz;
+                vertexNormal = (view * model * vec4(normal, 0.0)).xyz;
+                vertexColor = color;
+                gl_Position = projection * vec4(vertexPosition, 1.0);
+            }
+        `,
+        frag: `
+            precision mediump float;
+
+            const int MAX_LIGHTS = 1; // TODO: increase this and have a way to pass in lights
```

Comment:
Nit: do you mind adding your name after the TODO so we can quickly find out who to ask follow up questions about this TODO if we need it: like `TODO(dave): [...]`

Reply:
#10

Reply:
Added